src/Context/HTMLContext.php

```diff
@@ -63,4 +65,33 @@ public function thePageHtmlMarkupShouldNotBeValid(): void
             'HTML markup should not be valid.'
         );
     }
+
+    /**
+     * @Then the page HTML5 doctype declaration should be valid
+     */
+    public function thePageHtml5DoctypeDeclarationShouldBeValid(): void
+    {
+        Assert::eq(
+            self::HTML5_DOCTYPE_MARKUP,
+            $this->getFirstLineOfPageContent()
+        );
+    }
+
+    /**
+     * @Then the page HTML5 doctype declaration should not be valid
+     */
+    public function thePageHtml5DoctypeDeclarationShouldNotBeValid(): void
+    {
+        Assert::notEq(
+            self::HTML5_DOCTYPE_MARKUP,
+            $this->getFirstLineOfPageContent()
+        );
+    }
+
+    private function getFirstLineOfPageContent(): string
+    {
+        $pageContent = $this->getSession()->getPage()->getContent();
+        $pageContentLines = explode(PHP_EOL, $pageContent);
```

Comment:
According to the HTML specifications, whitespace and comments may come before the ``doctype`` declaration. With the current implementation, the library may throw false positives.